airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java

```diff
@@ -0,0 +1,130 @@
+  private final List<String> columnNames;
+  private final AirbyteStreamNameNamespacePair pair;
+  private final JdbcDatabase database;
+  // Represents the number of rows to get with each query.
+  private final long chunkSize;
+  private final PrimaryKeyInfo pkInfo;
+  private int numSubqueries = 0;
+  private AutoCloseableIterator<JsonNode> currentIterator;
+
+  MySqlInitialLoadRecordIterator(
+      final JdbcDatabase database,
+      final MySqlInitialLoadSourceOperations sourceOperations,
+      final String quoteString,
+      final MySqlInitialLoadStateManager initialLoadStateManager,
+      final List<String> columnNames,
+      final AirbyteStreamNameNamespacePair pair,
+      final long chunkSize) {
+    this.database = database;
+    this.sourceOperations = sourceOperations;
+    this.quoteString = quoteString;
+    this.initialLoadStateManager = initialLoadStateManager;
+    this.columnNames = columnNames;
+    this.pair = pair;
+    this.chunkSize = chunkSize;
+    this.pkInfo = initialLoadStateManager.getPrimaryKeyInfo(pair);
+  }
+
+  @CheckForNull
+  @Override
+  protected JsonNode computeNext() {
+    if (currentIterator == null || !currentIterator.hasNext()) {
+      try {
+        LOGGER.info("Subquery number : {}", numSubqueries);
+        final Stream<JsonNode> stream = database.unsafeQuery(
+            connection -> getPkPreparedStatement(connection), sourceOperations::rowToJson);
+
+        // Previous stream (and connection) must be manually closed in this iterator.
+        if (currentIterator != null) {
+          currentIterator.close();
+        }
+        currentIterator = AutoCloseableIterators.fromStream(stream, pair);
+        numSubqueries++;
+      } catch (final Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+    if (!currentIterator.hasNext()) {
+      return endOfData();
+    }
+    return currentIterator.next();
+  }
+
+  private PreparedStatement getPkPreparedStatement(final Connection connection) {
+    try {
+      final String tableName = pair.getName();
+      final String schemaName = pair.getNamespace();
+      LOGGER.info("Preparing query for table: {}", tableName);
+      final String fullTableName = getFullyQualifiedTableNameWithQuoting(schemaName, tableName,
+          quoteString);
+
+      final String wrappedColumnNames = RelationalDbQueryUtils.enquoteIdentifierList(columnNames, quoteString);
+
+      final PrimaryKeyLoadStatus pkLoadStatus = initialLoadStateManager.getPrimaryKeyLoadStatus(pair);
+
+      if (pkLoadStatus == null) {
+        LOGGER.info("pkLoadStatus is null");
+        final String quotedCursorField = enquoteIdentifier(pkInfo.pkFieldName(), quoteString);
+        final String sql = String.format("SELECT %s FROM %s ORDER BY %s LIMIT %s", wrappedColumnNames, fullTableName,
+            quotedCursorField, chunkSize);
+        final PreparedStatement preparedStatement = connection.prepareStatement(sql);
```

Comment:
Can we add a LOGGER.info line here to indicate query execution similar to L116?

Reply:
Yes! Good eye. We'll definitely need visibility there